frontend/apps/app/app/api/pglite/query/route.ts

```diff
@@ -0,0 +1,231 @@
+}
+
+function parseInsert(
+  sql: string,
+): { tableName: string; values: unknown[] } | null {
+  const insertRegex = /INSERT\s+INTO\s+(\w+)\s+VALUES\s*\(([^)]+)\)/i
+  const match = sql.match(insertRegex)
+
+  if (!match) return null
+
+  const tableName = match[1]
+  const valuesStr = match[2]
+  const values = valuesStr.split(',').map((val) => {
+    const trimmed = val.trim()
+    if (trimmed.startsWith("'") && trimmed.endsWith("'")) {
+      return trimmed.slice(1, -1)
+    }
+    if (!Number.isNaN(Number(trimmed))) {
+      return Number(trimmed)
+    }
+    return trimmed
+  })
+
+  return { tableName, values }
+}
+
+function parseSelect(
+  sql: string,
+): { tableName: string; columns: string[] } | null {
+  const selectRegex = /SELECT\s+(.+?)\s+FROM\s+(\w+)/i
+  const match = sql.match(selectRegex)
+
+  if (!match) return null
+
+  const columnsStr = match[1].trim()
+  const tableName = match[2]
+  const columns =
+    columnsStr === '*' ? ['*'] : columnsStr.split(',').map((col) => col.trim())
+
+  return { tableName, columns }
+}
+
+interface SqlExecutionResult {
+  command: string
+  rowCount: number
+  rows?: unknown[]
+}
+
+function executeSQL(sessionData: SessionData, sql: string): SqlExecutionResult {
+  const trimmedSql = sql.trim()
+
+  if (trimmedSql.toUpperCase().startsWith('CREATE TABLE')) {
+    const parsed = parseCreateTable(trimmedSql)
+    if (parsed) {
+      sessionData.tables[parsed.tableName] = {
+        columns: parsed.columns,
+        rows: [],
+      }
+      return { command: 'CREATE', rowCount: 0 }
+    }
+    throw new Error('Invalid CREATE TABLE syntax')
+  }
+
+  if (trimmedSql.toUpperCase().startsWith('INSERT INTO')) {
+    const parsed = parseInsert(trimmedSql)
+    if (parsed && sessionData.tables[parsed.tableName]) {
+      const table = sessionData.tables[parsed.tableName]
+      const columnNames = Object.keys(table.columns)
+      const row: { [key: string]: unknown } = {}
+
+      for (const [index, colName] of columnNames.entries()) {
+        row[colName] = parsed.values[index] || null
+      }
+
+      table.rows.push(row)
+      return { command: 'INSERT', rowCount: 1 }
+    }
+    throw new Error('Table does not exist or invalid INSERT syntax')
+  }
+
+  if (trimmedSql.toUpperCase().startsWith('SELECT')) {
+    const parsed = parseSelect(trimmedSql)
+    if (parsed && sessionData.tables[parsed.tableName]) {
+      const table = sessionData.tables[parsed.tableName]
+      const rows = table.rows.map((row) => {
+        if (parsed.columns.includes('*')) {
+          return row
+        }
+        const filteredRow: { [key: string]: unknown } = {}
+        for (const col of parsed.columns) {
+          if (Object.hasOwn(row, col)) {
+            filteredRow[col] = row[col]
+          }
+        }
+        return filteredRow
+      })
+      return { command: 'SELECT', rowCount: rows.length, rows }
+    }
+    throw new Error('Table does not exist or invalid SELECT syntax')
+  }
```

Comment:
Q: Are `JOIN` queries supported? For example: `SELECT * FROM table_a JOIN table_b ON table_a.id = table_b.a_id;`